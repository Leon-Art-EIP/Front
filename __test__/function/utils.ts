import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
