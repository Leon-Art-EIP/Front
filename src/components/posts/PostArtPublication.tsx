import Link from "next/link";
import { imageApi } from "../../tools/variables";

interface IPostArtPublicationProps {
  artPublication: {
    _id: string;
    image: string | undefined;
  };
}

export default function PostArtPublication(props: IPostArtPublicationProps): JSX.Element {
  let children: JSX.Element;

  if (props.artPublication) {
    children = (
      <img
        alt="art"
        src={`${imageApi}/${props.artPublication.image}`}
        className="h-32 rounded-2xl object-cover w-full"
      />
    );
  } else {
    children = <p>Voir publication</p>;
  }

  return <Link href={`/single/${props.artPublication._id}`}>{children}</Link>;
}
