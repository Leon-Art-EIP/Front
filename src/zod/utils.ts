export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
