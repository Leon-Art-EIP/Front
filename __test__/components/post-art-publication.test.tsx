import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import PostArtPublication from "../../src/components/posts/PostArtPublication";

test("Post deleteLoading snapshot", () => {
  testSnapshot(
    <PostArtPublication
      artPublication={{
        _id: "1",
        image: "image.jpg",
      }}
    />
  );
});
