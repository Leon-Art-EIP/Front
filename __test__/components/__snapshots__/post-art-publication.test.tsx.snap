// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post deleteLoading snapshot 1`] = `
<a
  href="/single/1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <img
    alt="art"
    className="h-32 rounded-2xl object-cover w-full"
    src="undefined/api/image.jpg"
  />
</a>
`;
